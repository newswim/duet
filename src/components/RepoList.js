import React, { Component } from 'react'
import GitHub from 'github-api'
import Repo from './Repo'

class RepoList extends Component {
  constructor (props) {
    super(props)
    this.state = {}
  }

  componentWillMount () {
    const gh = new GitHub()
    const repos = [
      'jekyll/jekyll',
      'jekyll/plugins',
      'jekyll/jekyll-admin',
      'jekyll/minima'
    ]
    this.setState({
      repos: repos.map(repo => gh.getRepo(...repo.split('/'))),
      issues: repos.map(repo => gh.getIssues(...repo.split('/')))
    })
    console.log(this.state.repos)
  }

  render () {
    return (
      <div>
        {this.state.repos.map((repo, i) => {
          return (
            <Repo key={i} repo={repo} issues={this.state.issues[i]} />
          )
        })}
      </div>
    )
  }
}

export default RepoList
