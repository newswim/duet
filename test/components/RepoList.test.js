import React from 'react'
import RepoList from '../../src/components/RepoList'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

const repos = [
  {
    "path": "github/hub",
    "description": "aaaa"
  },
  {
    "path": "github/linguist",
    "description": "cool"
  }
]

test('it renders the component tree', () => {
  const tree = renderer.create(
    <RepoList repos={repos} />
  )
  expect(tree).toMatchSnapshot()
})

test('it renders the correct amount of repos', () => {
  const repos1 = repos.concat({
    "path": "github/dmca",
    "description": "bbbb"
  })
  const wrapper1 = shallow(<RepoList repos={repos1} />)
  expect(wrapper1.find('Repo').length).toBe(3)

  const repos2 = repos.slice(0, 1)
  const wrapper2 = shallow(<RepoList repos={repos2} />)
  expect(wrapper2.find('Repo').length).toBe(1)
})
