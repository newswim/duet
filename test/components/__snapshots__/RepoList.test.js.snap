// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders the component tree 1`] = `
<div>
  <div
    className="Gx"
    style={
      Object {
        "boxSizing": "border-box",
        "display": "inline-block",
        "maxWidth": "100%",
        "minWidth": "50%",
        "verticalAlign": "top",
        "width": "calc(262144px - 51200%)",
      }
    }
  >
    <a
      className="jnqCsK"
      href={undefined}
    >
      <h2
        className="jWDnSI"
      />
    </a>
    <span
      className="eUzqvD"
    >
      ★ 
    </span>
    <p
      className="knmMnU"
    >
      aaaa
    </p>
    <br />
    <i>
      Currently there are no open issues, check back later!
    </i>
  </div>
  <div
    className="Gx"
    style={
      Object {
        "boxSizing": "border-box",
        "display": "inline-block",
        "maxWidth": "100%",
        "minWidth": "50%",
        "verticalAlign": "top",
        "width": "calc(262144px - 51200%)",
      }
    }
  >
    <a
      className="jnqCsK"
      href={undefined}
    >
      <h2
        className="jWDnSI"
      />
    </a>
    <span
      className="eUzqvD"
    >
      ★ 
    </span>
    <p
      className="knmMnU"
    >
      cool
    </p>
    <br />
    <i>
      Currently there are no open issues, check back later!
    </i>
  </div>
</div>
`;
